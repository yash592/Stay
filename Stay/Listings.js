import React, { Component, PureComponent } from "react";
import {
  Platform,
  StyleSheet,
  Text,
  View,
  Image,
  ScrollView,
  FlatList
} from "react-native";
import { ApolloProvider } from "react-apollo";
import { ApolloClient } from "apollo-client";
import { HttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import { Query } from "react-apollo";
import { ReservationCard } from "./src/components/common/ReservationCard";

const query = gql`
  query {
    reservations {
      name
      hotelName
      arrivalDate
      departureDate
    }
  }
`;

class Listings extends PureComponent {
  renderReservations = ({ item }) => {
    console.log(this.props.data.loading);

    return (
      <ReservationCard
        name={item.name}
        hotelName={item.hotelName}
        arrivalDate={item.arrivalDate}
        departureDate={item.departureDate}
      />
    );
  };

  render() {
    console.log(this.props);
    if (this.props.data.loading) {
      return (
        <View
          style={{ flex: 1, justifyContent: "center", alignItems: "center" }}
        >
          <Text style={{ fontSize: 46 }}>Loading</Text>
        </View>
      );
    }

    return (
      <View>
        <FlatList
          data={this.props.data.reservations}
          initialNumToRender={7}
          renderItem={this.renderReservations}
        />
      </View>
    );
  }
}

export default graphql(query)(Listings);
