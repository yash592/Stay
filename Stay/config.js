export const END_POINT =
  "https://us1.prisma.sh/public-luckox-377/reservation-graphql-backend/dev";
