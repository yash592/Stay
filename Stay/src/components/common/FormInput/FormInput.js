import React, { Component } from "react";
import { View, Text, TextInput } from "react-native";

const FormInput = props => (
  <View style={styles.containerStyle}>
    <Text>{props.text}</Text>
    <TextInput
      style={styles.textInput}
      placeholder={props.placeholder}
      placeholderTextColor={props.placeholderTextColor}
    />
  </View>
);

const styles = {
  textInput: {
    width: "100%",
    backgroundColor: "white",
    height: 40,
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 10,
    padding: 10
  },
  containerStyle: {
    width: "42%",
    height: 50,
    margin: 5
  }
};
export { FormInput };
