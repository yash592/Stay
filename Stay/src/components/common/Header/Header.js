import React from "react";
import { Text, View } from "react-native";

// reusable header sticky component with title passed down as a prop

const Header = props => {
  const { textStyle, viewStyle } = styles;
  const { headerText } = props;

  return (
    <View style={viewStyle}>
      <Text style={textStyle}>{headerText}</Text>
    </View>
  );
};

const styles = {
  viewStyle: {
    backgroundColor: "#F8F8F8",
    justifyContent: "center",
    alignItems: "center",
    height: 60,
    width: "100%",

    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    elevation: 2,
    position: "relative",
    backgroundColor: "#D0FFC8"
  },
  textStyle: {
    fontSize: 20,
    fontFamily: "sans-serif-condensed",
    color: "black"
  }
};

// Make the component available to other parts of the app
export { Header };
