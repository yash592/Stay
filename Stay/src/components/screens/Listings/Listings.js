import React, { Component, PureComponent } from "react";
import {
  Platform,
  StyleSheet,
  Text,
  View,
  Image,
  ScrollView,
  FlatList,
  TouchableOpacity
} from "react-native";
import { ApolloProvider } from "react-apollo";
import { ApolloClient } from "apollo-client";
import { HttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import { Query } from "react-apollo";
import { ReservationCard } from "../../common/ReservationCard";
import { BottomNav } from "../../common/BottomNav";
import { Header } from "../../common/Header";
import { Loading } from "../../common/Loading";
import { Scene, Router, Actions } from "react-native-router-flux";
import { getAllReservations } from "../../../queries/queries.js";

// Listings component to display all reservations as the default screen

class Listings extends Component {
  // renderReservations function for the flatList
  // -- returns the ReservationCard component and passes it the props

  keyExtractor = (item, index) => item.id;

  renderReservations = ({ item }) => {
    // console.log(item);
    return (
      <View
        style={{
          alignItems: "center",
          justifyContent: "center"
        }}
      >
        <ReservationCard
          name={item.name}
          hotelName={item.hotelName}
          arrivalDate={item.arrivalDate}
          departureDate={item.departureDate}
        />
      </View>
    );
  };

  render() {
    const { loading, reservations } = this.props.data;

    // simple loading component to let the user know that the screen is loading

    if (loading) {
      return <Loading />;
    }

    return (
      <View style={{ flex: 1 }}>
        <Header headerText={"RESERVATIONS"} />

        <FlatList
          data={reservations}
          initialNumToRender={3}
          renderItem={this.renderReservations}
          keyExtractor={this.keyExtractor}
          initialNumToRender={8}
        />

        <BottomNav />
      </View>
    );
  }
}

export default graphql(getAllReservations)(Listings);
