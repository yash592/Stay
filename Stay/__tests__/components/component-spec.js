/**
 * @format
 */

import "react-native";
import renderer from "react-test-renderer";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";

configure({ adapter: new Adapter() });
import Listings from "../../src/components/screens/Listings/Listings";
import { FormInput } from "../../src/components/common/FormInput";
import { Header } from "../../src/components/common/Header";
import { shallow } from "enzyme";

import { END_POINT } from "../../config";
import { MockedProvider } from "react-apollo/test-utils";
import {
  getAllReservations,
  getOneReservation
} from "../../src/queries/queries.js";
// import { mocks } from "./mocks";
const wait = require("waait");

// snapshot of the button component

describe("FormInput component renders correctly", () => {
  it("Shallow rendering", () => {
    const component = shallow(<FormInput />);
    expect(component).toMatchSnapshot();
  });
});

// snapshot of the Header common components

describe("Header component renders correctly", () => {
  it("Shallow rendering", () => {
    const component = shallow(<Header headerText="Test" />);
    expect(component).toMatchSnapshot();
  });
});

// snapshot of the Listings component

describe("Listing component renders correctly", () => {
  it("Shallow rendering", () => {
    const component = shallow(<Listings />);
    expect(component).toMatchSnapshot();
  });
});
