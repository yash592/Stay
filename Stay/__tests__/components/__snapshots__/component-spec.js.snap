// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component renders correctly Shallow rendering 1`] = `
<View
  style={
    Object {
      "height": 50,
      "margin": 5,
      "width": "92%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#d1d7db",
        "borderWidth": 1,
        "height": 40,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Header component renders correctly Shallow rendering 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D0FFC8",
      "elevation": 2,
      "height": 60,
      "justifyContent": "center",
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "sans-serif-condensed",
        "fontSize": 20,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Listing component renders correctly Shallow rendering 1`] = `
<Query
  displayName="Apollo(Listings)"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "orderBy",
                    },
                    "value": Object {
                      "kind": "EnumValue",
                      "value": "createdAt_DESC",
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "reservations",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "hotelName",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "arrivalDate",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "departureDate",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 139,
        "start": 0,
      },
    }
  }
  skip={false}
  warnUnhandledError={true}
>
  <Component />
</Query>
`;
